### PR TITLE
Move TF2 Keras layer tests from private repo and adapt them

### PR DESCRIPTION
**Pull request by gkrivor**
**Details:** Moving the TF2 layer tests from the private repository and adapting them to be run in the scope of the public repo.

Moving layer tests for TensorFlow2 Keras
Fixed behavior of use_new_framework flag
Fixed handling of transpose operation in case of OV2.0

### CODE REVIEW

**Comment by rkazants** [comment]
@gkrivor, any progress to make CI green and apply the feedback?

**Comment by gkrivor** [comment]
> @gkrivor, any progress to make CI green and apply the feedback?

CI is unstable, I think question should be addressed to CI team, I can only push "Rebuild" from my side.

I want to get at least one green CI before enabling to be sure it isn't caused by added tests. I thought to add it as a separate PR, if you want to add it in this PR - I'm ok.

**Comment by rkazants** [comment]
> > @gkrivor, any progress to make CI green and apply the feedback?
> 
> CI is unstable, I think question should be addressed to CI team, I can only push "Rebuild" from my side.
> 
> I want to get at least one green CI before enabling to be sure it isn't caused by added tests. I thought to add it as a separate PR, if you want to add it in this PR - I'm ok.

The current instability of CI should not affect turning on some TF2 layers tests at all. It is mostly related to plugins etc, MO is stable and generate the determined IRs.
Please turn on a few TF2 layer tests (just one simple test) in the precommit on CPU, for example. This way we guarantee that TF2 is moved correctly here.

**Comment by gkrivor** [comment]
Tried to add a test. Let's take a look how it works...

**Review by rkazants — APPROVED** [review]

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -436,7 +436,15 @@ jobs:
...
       $(RUN_PREFIX) python3 -m pytest $(LAYER_TESTS_DIR)/tensorflow_tests/test_tf_Roll.py --ir_version=10 --junitxml=$(INSTALL_TEST_DIR)/TEST-tf_Roll.xmlTEST
-    displayName: 'Layer Tests'
+    displayName: 'Layer Tests - Tensorflow'
+    continueOnError: false
+
+  - script: |
+      python3 -m pip install -r $(LAYER_TESTS_DIR)/requirements.txt
+      export PYTHONPATH=$(LAYER_TESTS_DIR):$PYTHONPATH
+      export TEST_DEVICE=CPU
+      $(RUN_PREFIX) python3 -m pytest $(LAYER_TESTS_DIR)/tensorflow2_keras_tests/test_tf2_keras_activation.py --ir_version=11 --junitxml=./TEST-tf2_Activation.xmlTEST -k "sigmoid"
```
Comment (rkazants) [inline .ci/azure/linux.yml:446]: Please change this line to run all pre-commit cases in a separate task/PR. Of course, the pre-commit cases should be revised first - let us say to have one pre-commit basic case per operation.

**Review by rkazants — APPROVED** [review]

**Pull request merged**